### PR TITLE
Add callback_url/acct information for Sidekiq PuSH workers Exception.

### PR DESCRIPTION
**Pull request by clworld**
Some exception does not report hostname in exception message.
So, hostname does not shown in Sidekiq error console.
You need check database tables to know which host returns error.
With this PR, you can easily see which server is in trouble...
sidekiq_fix3

### CODE REVIEW

**Review by beatrix-bitrot — APPROVED** [review]
YES PLEASE

**Comment by beatrix-bitrot** [comment]
ok maybe fix the broken tests d:

still... i really want this

File: app/lib/exceptions.rb [hunk 1]
```diff
@@ -8,11 +8,11 @@ class RaceConditionError < Error; end
 
   class UnexpectedResponseError < Error
     def initialize(response = nil)
-      @response = response
-    end
-
-    def to_s
-      "#{@response.uri} returned code #{@response.code}"
+      if response.respond_to? :url
+        super("#{response.uri} returned code #{response.code}")
```
Comment (nightpool) [inline app/lib/exceptions.rb]: url vs uri?

Comment (clworld) [inline app/lib/exceptions.rb]: Oh. mistake…

**Comment by nightpool** [comment]
What is the difference between "response.uri" and "acct"? One gives more meaningful information, in my opinion.

**Comment by nightpool** [comment]
This feels like it's redundant now that we use MastodonUnexpectedResponse.

**Comment by clworld** [comment]
I'm confused about RSpec executes Sidekiq worker directly and exceptions thrown by workers returns to caller...
I choose way that don't change Exception class. but this way assumes all StandardError are reproducible..
Another way is wrap errors by some Exception (Mastodon::WorkerError). It breaks rspec which checks which exception is raised.

**Comment by clworld** [comment]
@nightpool 
responce.uri is what UnexpectedResponseError show on .message.
I don't change UnexpectedResponseError's output. I changed UnexpectedResponseError to make it reproducible.

What I changed in this PR is add acct/callback_url to Exceptions which Sidekiq workers raises.
callback_url is url to post OStatus payload.
acct is Mastodon account name to subscribe. For example: "clworld@md.ggtea.org" is my account's acct.

**Comment by nightpool** [comment]
okay, that makes sense. LGTM

**Review by unarist — COMMENTED** [review]
Looks good, but I'm not sure about `raise e.class "..."` is proper way or not.

**Comment by clworld** [comment]
I send alternative implementation: https://github.com/tootsuite/mastodon/pull/4283
which don't use 'raise e.class "..."' thing. but wrap all exceptions.

**Review by Gargron — APPROVED** [review]

**Pull request merged**